README.md

```diff
@@ -10,14 +10,101 @@ and ease of use. Try our online demos:
 [LLaMA2](https://huggingface.co/spaces/lmz/candle-llama2),
 [yolo](https://huggingface.co/spaces/lmz/candle-yolo).
 
+## Installation
+
+- **With Cuda support**:
+
+1. First, make sure that Cuda is correctly installed.
+- `nvcc --version` should print your information about your Cuda compiler driver.
+- `nvidia-smi --query-gpu=compute_cap --format=csv` should print your GPUs compute capability, e.g. something
+like:
+```
+compute_cap
+8.9
+```
+
+If any of the above commands errors out, please make sure to update your Cuda version.
+
+2. Create a new app and add [`candle-core`](https://github.com/huggingface/candle/tree/main/candle-core) with Cuda support
+
+Start by creating a new cargo:
+
+```bash
+cargo new myapp
+cd myapp
+```
+
+Make sure to add the `candle-core` crate with the cuda feature:
+
+```
+cargo add --git https://github.com/huggingface/candle.git candle-core --features "cuda"
+```
+
+Run `cargo build` to make sure everything can be correctly built.
+
+```
+cargo run
+```
+
+**Without Cuda support**:
+
+Create a new app and add [`candle-core`](https://github.com/huggingface/candle/tree/main/candle-core) as follows:
+
+```
+cargo new myapp
+cd myapp
+cargo add --git https://github.com/huggingface/candle.git candle-core
+```
+
+Run `cargo build` to make sure everything can be correctly built.
+
+```
+cargo run
+```
+
+## Get started
+
+Having installed `candle-core` as described in [Installation](#Installation), we can now 
+run a simple matrix multiplication.
+
+We will need the [`anyhow`](https://docs.rs/anyhow/latest/anyhow/) package for our example,
+so let's add it to our app.
+
+```
+cd myapp
+cargo add anyhow
+```
+
+Next, write the following to your `myapp/src/main.rs` file:
+
 ```rust
-let a = Tensor::randn(0f32, 1., (2, 3), &Device::Cpu)?;
-let b = Tensor::randn(0f32, 1., (3, 4), &Device::Cpu)?;
+use anyhow::Result;
+use candle_core::{Device, Tensor};
+
+fn main() -> Result<()> {
+    let device = Device::Cpu;
```

Comment:
changed the example this way so that one by default only creates one device.

Reply:
Thanks, for the `Cpu` one it doesn't matter much as the enum case doesn't embed any field but seems good to show the good practice here in a consistent way across devices.